dogma.go

```diff
@@ -215,29 +215,28 @@ func (c *Client) do(ctx context.Context, req *http.Request, resContent interface
 
 	res, err := c.client.Do(req)
 	if err != nil {
-		select {
-		case <-ctx.Done():
-			return nil, ctx.Err()
```

Comment:
the context's error will be probably more useful?

Reply:
When context is cancelled or error, `res, err` would hold the error. So we don't need `ctx.Err()` here

Reply:
There's a chance that it gets an error response and the user cancels the `ctx` before the code reaches here.
Please see https://github.com/google/go-github/blob/master/github/github.go#L484 this.
But it's probably no big deal so we can just return the original `err`.

Reply:
Yes, in case user cancel the `ctx` first, then it's his choice to cancel whole process and stop. I think it's no big deal too.